spring-integration-core/src/main/java/org/springframework/integration/router/AbstractMappingMessageRouter.java

```diff
@@ -70,9 +68,8 @@
 	@ManagedAttribute
 	public void setChannelMappings(Map<String, String> channelMappings) {
 		Assert.notNull(channelMappings, "'channelMappings' must not be null");
-		Map<String, String> newChannelMappings = new ConcurrentHashMap<String, String>();
-		newChannelMappings.putAll(channelMappings);
-		this.doSetChannelMappings(newChannelMappings);
+		Map<String, String> newChannelMappings = new ConcurrentHashMap<String, String>(channelMappings);
+		doSetChannelMappings(newChannelMappings);
```

Comment:
??? Sorry, I am missing anything, or just don't understand you...
```
private void doSetChannelMappings(Map<String, String> newChannelMappings) {
	Map<String, String> oldChannelMappings = this.channelMappings;
	this.channelMappings = newChannelMappings;
	if (logger.isDebugEnabled()) {
		logger.debug("Channel mappings: " + oldChannelMappings + " replaced with: " + newChannelMappings);
	}
}
```
We replace the old field value with the new one. There is no any `clear`.
We use this `doSetChannelMappings()` in those cases when we fully replace one mapping set with another, like this `setChannelMappings()`. 
We can logging for others, but that does not make so much sense.
From other side we even can get rid of this logging as well: you always can use `getChannelMappings()` for diagnostics before setting a new one.